actionpack/lib/action_dispatch/middleware/stack.rb

```diff
@@ -130,7 +130,11 @@ def swap(target, *args, &block)
     ruby2_keywords(:swap)
 
     def delete(target)
-      middlewares.reject! { |m| m.name == target.name } || (raise "No such middleware to delete: #{target.inspect}")
+      middlewares.reject! { |m| m.name == target.name }
```

Comment:
The original implementation used `delete_if` instead of `reject!`.
`delete_if` returns the `middlewares` if nothing is found.
`reject!` returns nil if nothing is found.